src/services/completions.ts

```diff
@@ -658,6 +670,230 @@ function isCheckedFile(sourceFile: SourceFile, compilerOptions: CompilerOptions)
+                if (enumValue !== undefined) {
+                    if (tracker.hasValue(enumValue)) {
+                        continue;
+                    }
+                    tracker.addValue(enumValue);
+                }
+                const typeNode = codefix.typeToAutoImportableTypeNode(checker, importAdder, type, caseBlock, target);
+                if (!typeNode) {
+                    return undefined;
+                }
+                const expr = typeNodeToExpression(typeNode, target, quotePreference);
+                if (!expr) {
+                    return undefined;
+                }
+                elements.push(expr);
+            }
+            // Literals
+            else if (!tracker.hasValue(type.value)) {
+                switch (typeof type.value) {
+                    case "object":
+                        elements.push(factory.createBigIntLiteral(type.value));
+                        break;
+                    case "number":
+                        elements.push(factory.createNumericLiteral(type.value));
+                        break;
+                    case "string":
+                        elements.push(factory.createStringLiteral(type.value));
+                        break;
+                }
+            }
+        }
+        if (elements.length === 0) {
+            return undefined;
+        }
+
+        const newClauses = map(elements, element => {
+            return factory.createCaseClause(element, []);
+        });
+        const printer = createSnippetPrinter({
+            removeComments: true,
+            module: options.module,
+            target: options.target,
+            newLine: getNewLineKind(getNewLineCharacter(options, maybeBind(host, host.getNewLine))),
+        });
+        const insertText = formatContext
+            ? printer.printAndFormatSnippetList(
+                ListFormat.MultiLine | ListFormat.NoTrailingNewLine,
+                factory.createNodeArray(newClauses),
+                sourceFile,
+                formatContext)
+            : printer.printSnippetList(
+                ListFormat.MultiLine | ListFormat.NoTrailingNewLine,
+                factory.createNodeArray(newClauses),
+                sourceFile);
+
+        const firstClause = printer.printSnippetList(ListFormat.SingleLine, factory.createNodeArray([first(newClauses)!]), sourceFile);
+        return {
+            entry: {
+                name: `${firstClause} ...`,
+                kind: ScriptElementKind.unknown,
+                sortText: SortText.GlobalsOrKeywords,
+                insertText,
+                hasAction: importAdder.hasFixes() || undefined,
+                source: CompletionSource.SwitchCases,
+            },
+            importAdder,
+        };
+    }
+
+    return undefined;
+}
+
+interface CaseClauseTracker {
+    addClause(clause: CaseClause): void;
+    addValue(value: string | number): void;

```

Comment:
As it is today, we don't yet need to add a BigInt with `addValue`, because we add them when processing the existing case clauses. We only use `addValue` so far for adding enum member values, and those can't be BigInts. I figured I'd only implement what I'm currently using.

Reply:
I guess on second thought, I could refactor things to always use `addValue`. I think I'll do that.